com.unity.multiplayer.transport.utp/Runtime/UTPTransport.cs

```diff
@@ -173,236 +174,196 @@ public enum UTPDelivery
-            channelNameToId.Add(MLAPI_CHANNELS[i].Name, i);
-            internalChannels.Add(i, new UTPChannel() {
-                Id = i,
-                Name = MLAPI_CHANNELS[i].Name,
-                Flags = MLAPIChannelTypeToPacketFlag(MLAPI_CHANNELS[i].Type)
-            });
-        }
+        m_NetworkPipelines[0] = Driver.CreatePipeline(typeof(NullPipelineStage));
+        m_NetworkPipelines[1] = Driver.CreatePipeline(typeof(ReliableSequencedPipelineStage));
+        m_NetworkPipelines[2] = Driver.CreatePipeline(typeof(UnreliableSequencedPipelineStage));
 
-        m_packetData = new NativeQueue<RawNetworkMessage>(Allocator.Persistent);
-        m_packetProcessBuffer = new NativeArray<byte>(1000, Allocator.Persistent);
+        PacketData = new NativeQueue<RawNetworkMessage>(Allocator.Persistent);
+        m_PacketProcessBuffer = new NativeArray<byte>(1000, Allocator.Persistent);
     }
 
-    public UTPDelivery MLAPIChannelTypeToPacketFlag(ChannelType type)
+    [BurstCompile]
+    public void SendToClient(NativeArray<byte> packet, ulong clientId, int index)
     {
-        switch (type) {
-            case ChannelType.Unreliable: {
-                return UTPDelivery.Unreliable;
-            }
-            case ChannelType.Reliable: {
+        for (int i = 0; i < Connections.Length; i++) {
+            if (Connections[i].InternalId != (int)clientId)
+                continue;
 
-                return UTPDelivery.ReliableSequenced;
-            }
-            case ChannelType.ReliableSequenced: {
-                return UTPDelivery.ReliableSequenced;
-            }
-            case ChannelType.ReliableFragmentedSequenced: {
-                return UTPDelivery.ReliableSequenced;
-            }
-            case ChannelType.UnreliableSequenced: {
-                return UTPDelivery.UnreliableSequenced;
-            }
-            default: {
-                throw new ArgumentOutOfRangeException(nameof(type), type, null);
-            }
+            var writer = Driver.BeginSend(m_NetworkPipelines[index], Connections[i]);
+
+            if (!writer.IsCreated)
+                continue;
+
+            writer.WriteBytes(packet);
+
+            Driver.EndSend(writer);
         }
     }
 
-    public override NetEventType PollEvent(out ulong clientId, out string channelName, out ArraySegment<byte> payload, out float receiveTime)
+    public override unsafe void Send(ulong clientId, ArraySegment<byte> data, NetworkChannel networkChannel)
+    {
+        var pipelineIndex = 0;
+
+        GetUTPConnectionDetails(clientId, out uint peerId);
+
+        var writer = new DataStreamWriter(data.Count + 1 + 4, Allocator.Temp);
+        writer.WriteByte((byte)networkChannel);
+        writer.WriteInt(data.Count);
+
+        fixed (byte* dataArrayPtr = data.Array) {
+            writer.WriteBytes(dataArrayPtr, data.Count);
+        }
+
+        SendToClient(writer.AsNativeArray(), peerId, pipelineIndex);
```

Comment:
Should we just eliminate pipelineIndex here and in the SendToClient function?

Reply:
Yea as mentioned in the comment we are re-writing the this, but in order to move the STAR Document forward I need to be able to publish the package and pass validation and I also had to fix compile errors since this package wasn't in CI to begin with.

Reply:
ah, ok...I guess if this feature is disabled and we can't ship without it re-enabled it would seem worth a comment here and/or ticket targeted for 1.0.0 so we don't forget

Reply:
We have a ticket in the DOTS Server Jira to re-write all this code :P